src/client/requirements.js

```diff
@@ -11,14 +11,16 @@ export const DEFAULTS = {
  * @return {Object} New requirements endpoint tree.
  */
 export function combineComponentRequirements( requirementsByComponent ) {
-	const requirements = {};
+	const requirementsByEndpoint = {};
 
-	requirementsByComponent.forEach( ( requirement ) => {
-		const { endpoint, params, ...reqParams } = requirement;
-		addEndpointRequirement( requirements, reqParams, endpoint, params );
+	requirementsByComponent.forEach( ( requirements ) => {
```

Comment:
Do you think we need any validation here that `requirements` and `requirement` are an array?

Reply:
Well, given that this is internal code and `requirementsByComponent` is only returned by internal code here, I'd prefer to rely on unit tests and not add unnecessary validation to the distributed code package.